crates/store/re_grpc_client/src/connection_registry.rs

```diff
@@ -123,14 +157,61 @@ impl ConnectionRegistryHandle {
                 .or_else(get_token_from_env)
         };
 
-        let client = crate::redap::client(origin.clone(), token).await;
-        match client {
+        let client = crate::redap::client(origin.clone(), token.clone()).await;
+        let mut client = match client {
             Ok(client) => {
                 let mut inner = self.inner.write().await;
-                inner.clients.insert(origin, client.clone());
-                Ok(ConnectionClient::new(client))
+                inner.clients.insert(origin.clone(), client.clone());
+                ConnectionClient::new(client)
+            }
+            Err(err) => {
+                return Err(err.into());
             }
-            Err(err) => Err(err),
+        };
+
+        // Call the version endpoint to check that authentication is successful. It's ok to do this
+        // since we're caching the client, so we're not spamming such request unnecessarily.
+        let request_result = client
+            .inner()
+            .version(re_protos::frontend::v1alpha1::VersionRequest {})
```

Comment:
Discussed on slack: the version endpoint doesn't actually check for authentication. We need a `/whoami` endpoint for that:

https://rerunio.slack.com/archives/C07T0LFT5BQ/p1756219199757459